moses/baselines/ngram/ngram.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+import moses
+from moses import CharVocab
+from tqdm import tqdm
+
+
+class Model:
+    def __init__(self, dataset, context_len=3, l_smooth=0.1):
+        self.vocab = CharVocab.from_data(dataset)
+        self.n = context_len
+        self.n_tokens = len(self.vocab)
+        self.l_smooth = l_smooth
+        self._dict = dict()
+        self.default_probs = np.hstack([np.ones(self.n_tokens-4),
+                                        np.array([0., 1., 0., 0.])]) * l_smooth
+        self.fit(dataset)
+    
+    def fit(self, data):
+        with tqdm(total=len(data)) as pbar:
+            for i, line in enumerate(data):
```

Comment:
You can use tqdm directly around iterables: `tqdm(enumerate(data), total=len(data))`